sdk/helper/ldaputil/client.go

```diff
@@ -553,7 +558,7 @@ func (c *Client) GetLdapGroups(cfg *ConfigEntry, conn Connection, userDN string,
 	if cfg.UseTokenGroups {
 		entries, err = c.performLdapTokenGroupsSearch(cfg, conn, userDN)
 	} else {
-		if paging, ok := conn.(PagingConnection); ok && cfg.MaximumPageSize >= 0 {
+		if paging, ok := conn.(PagingConnection); ok && *cfg.MaximumPageSize >= 0 {
```

Comment:
I propose that if the page size is set to 0, a non-paged search should be done, as 0 is an invalid value to use as a page size.

```suggestion
		if paging, ok := conn.(PagingConnection); ok && *cfg.MaximumPageSize > 0 {
```

Reply:
That comment in go-ldap looks incorrect to me. I tested it, too, pointing Vault at OpenLDAP and monitoring the LDAP traffic with Wireshark - 0 resulted in the defective behaviour that many people have reported in #20416, not what that comment claims.

Reply:
Thanks @maxb! My concern here is different flavors of LDAP may have different results but looking at [LDAP RFC's](https://docs.ldap.com/specs/rfc2696.txt) I don't really see anything covering this except:

```
   A sequence of paged search requests is abandoned by the client
   sending a search request containing a pagedResultsControl with the
   size set to zero (0) and the cookie set to the last cookie returned
   by the server.  A client MAY use the LDAP Abandon operation to
   abandon one paged search request in progress, but this is discouraged
   as it MAY invalidate the client's cookie.
```

@ltcarbonell I think we can go with 0.